functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -0,0 +1,224 @@
+ */
+package com.redhat.che.selenium.core.workspace;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Request.Builder;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Anatolii Bazko
+ */
+public class CheStarterWrapper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CheStarterWrapper.class);
+
+  private String host;
+  private String osioUrlPart;
+  private String cheStarterURL = "http://localhost:10000";
+
+  @Inject
+  public CheStarterWrapper(
+      @Named("che.osio.url") String osioUrlPart,
+      @Named("che.host") String cheHost,
+      @Named("che.chromedriver.port") String chromedriverPort
+  ) throws IOException, InterruptedException {
+    this.host = cheHost;
+    this.osioUrlPart = osioUrlPart;
+    /* RUN CHROMEDRIVER */
+    String chromeDriverCheckCommand =
+        "lsof -i TCP | grep -q 'localhost:" + chromedriverPort + " (LISTEN)'";
+    Process chromeDriverCheck = Runtime.getRuntime().exec(chromeDriverCheckCommand);
+    chromeDriverCheck.waitFor();
+    if (chromeDriverCheck.exitValue() != 0) {
+      try {
+        Process chromedriver = Runtime.getRuntime().exec("chromedriver");
+        LOG.info("Chromedriver successfully started.");
+      } catch (IOException e) {
+        LOG.error("Could not start process chromedriver:" + e.getMessage());
+        throw e;
+      }
+    }
+  }
+
+  public void start() throws IllegalStateException {
+    //TODO: Check if che starter is running;
+    try {
+      File cheStarterDir =
+          new File(System.getProperty("user.dir"), "target" + File.separator + "che-starter");
+
+      cloneGitDirectory(cheStarterDir);
+
+      LOG.info("Running che starter.");
+      Properties props = new Properties();
+      props.setProperty(
+          "OPENSHIFT_TOKEN_URL",
+          "https://sso." + this.osioUrlPart + "/auth/realms/fabric8/broker/openshift-v3/token");
+      props.setProperty(
+          "GITHUB_TOKEN_URL",
+          "https://auth." + this.osioUrlPart + "/api/token?for=https://github.com");
+      props.setProperty(
+          "CHE_SERVER_URL", "https://rhche." + this.osioUrlPart);
+      String pom = cheStarterDir.getAbsolutePath() + File.separator + "pom.xml";
```

Comment:
I think it is better to start che starter as an image then do it programmatically from test code. This approach is easier to maintain.

Reply:
I could try to discuss that with @Katka92 and figure out a solution :)

Reply:
Working on it.